### ISSUE TITLE
Language is not set when navigating back (client side) to the previous language

### ISSUE THREAD

**Issue by StarpTech** [isaachinman/next-i18next#559]
## Describe the bug
When I switch the language e.g from `/de/products/1` to `/en/products/1` by `i18n.changeLanguage` and use the browser back button to see the content again in `de` the language is not changed and therefore the content language wasn't requested (getInitialProps) in the correct language.

When I navigate multiple times back no `getInitialProps` is fired at all only the URL is changed.

## Expected behaviour
I would expect that the language is updated so that the content is fetched in the correct language.

## OS (please complete the following information)
 - Browser: [Chrome 70]

**Comment by isaachinman** [comment]
@StarpTech Is this resolved now with the shallow routing flag?

**Comment by StarpTech** [comment]
I can only reproduce it in development. For example.

1. Navigate to `en` content page
2. Throw an error in `de` content page and navigate to `de` content page
4. Click on browser back button
5. In network tab, you will see that the language is still `en` not `de` and even the current url is `/de/products/....`

**Comment by isaachinman** [comment]
Sorry, what does throwing an error have to do with it? Can you upload a short gif of this behaviour?

**Comment by StarpTech** [comment]
I think I got it. The issue is that the cookie is not updated on popstate events and my application will request content in a language that doesn't exist.
I can reproduce in development and production.

In that way I get the current lang:

```js
  axiosReq.params = {
    lng: serverReq ? serverReq.language : i18n.language,
    ...axiosReq.params
  };
```

**Comment by StarpTech** [comment]
@isaachinman when I use the back button to see the page in the previous lang the content is still fetched in the current lang. Did you understand?

**Comment by StarpTech** [comment]
@isaachinman I could fix it with

```
class MyApp extends App {
  componentDidMount() {
    Router.beforePopState(({ as }) => {
      for (let i = 0; i < i18n.options.allLanguages.length; i += 1) {
        const lng = i18n.options.allLanguages[i];

        if (as.startsWith(`/${lng}`)) {
          Cookies.set('lang', lng);
          break;
        }
      }

      return true;
    });
  }
```

**Comment by isaachinman** [comment]
Might just be a matter of moving `beforePopState` from `coreMethods` to `wrappedMethods` [here](https://github.com/isaachinman/next-i18next/blob/4e3f64958aa586b3b042ff15239e6abe2b0f00f1/src/router/wrap-router.ts#L38).

**Comment by StarpTech** [comment]
@isaachinman no idea how that should work because that [place](https://github.com/isaachinman/next-i18next/blob/4e3f64958aa586b3b042ff15239e6abe2b0f00f1/src/router/wrap-router.ts#L38) only corrects the paths and don't interact with the i18n instance.

**Comment by isaachinman** [comment]
@StarpTech Well, the problem here is that we've got a mismatch of URL and locale, right? I would assume in that case we should assume the source of truth is the URL, if it doesn't match with the current locale.

Keep in mind that wrapping `Router` methods is not the full story - we also have to support the `Link` component.

It's indeed a tricky problem. Let me know if you come up with any simple ideas to fix it.

**Comment by StarpTech** [comment]
> It's indeed a tricky problem. Let me know if you come up with any simple ideas to fix it.

The most simple trick is to do a hard reload when the user navigates through the client's history and the locale doesn't match with the URL locale. 

> Keep in mind that wrapping Router methods is not the full story - we also have to support the Link component.

In which way? From my understanding, we only need to change the global language in `beforePopState` without reloading `getInitialProps`. We need a way to do shallow routing selectively or expose the whole i18n instance to change the global change. WDYT?

**Comment by isaachinman** [comment]
Hi there, I have just released `next-i18next@8.0.0`, which uses the internationalised routing built into NextJs v10+. This new version of `next-i18next` is quite different, thus making this PR no longer relevant. Thanks!
